docs/howto/features/gpu.md

```diff
@@ -0,0 +1,140 @@
+
+   ```bash
+   jsonnet <your-cluster>.jsonnet > <your-cluster>.eksctl.yaml
+   ```
+
+3. Create the nodegroup
+
+   ```bash
+   eksctl create nodegroup -f <your-cluster>.eksctl.yaml --install-nvidia-plugin=false
+   ```
+
+   The `--install-nvidia-plugin=false` is required until
+   [this bug](https://github.com/weaveworks/eksctl/issues/5277)
+   is fixed.
+
+   This should create the nodegroup with 0 nodes in it, and the
+   autoscaler should recognize this!
+
+#### Setting up a GPU user profile
+
+Finally, we need to give users the option of using the GPU via
+a profile. This should be placed in the hub configuration:
+
+```yaml
+jupyterhub:
+    singleuser:
+        profileList:
+        - display_name: "Large + GPU: p2.xlarge"
+          description: "~4CPUs, 60G RAM, 1 NVIDIA K80 GPU"
+          kubespawner_override:
+            mem_limit: null
+            mem_guarantee: 55G
+            image: "pangeo/ml-notebook:<tag>"
+            environment:
+              NVIDIA_DRIVER_CAPABILITIES: compute,utility
+            extra_resource_limits:
+              nvidia.com/gpu: "1"
+            node_selector:
+              node.kubernetes.io/instance-type: p2.xlarge
+```
+
+1. If using a `daskhub`, place this under the `basehub` key.
+2. The image used should have ML tools (pytorch, cuda, etc)
+   installed. The recommendation is to use Pangeo's
+   [ml-notebook](https://hub.docker.com/r/pangeo/ml-notebook)
+   for tensorflow and [pytorch-notebook](https://hub.docker.com/r/pangeo/pytorch-notebook)
+   for pytorch. **Do not** use the `latest` or `master` tags - find
+   a specific tag listed for the image you want, and use that.
+3. The [NVIDIA_DRIVER_CAPABILITIES](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html#driver-capabilities)
+   environment variable tells the GPU driver what kind of libraries
+   and tools to inject into the container. Without setting this,
+   GPUs can not be accessed.
+4. The `node_selector` makes sure that these user pods end up on
+   the appropriate nodegroup we created earlier. Change the selector
+   and the `mem_guarantee` if you are using a different kind of node
+
+Do a deployment with this config, and then we can test to make sure
+this works!
+
+#### Testing
+
+1. Login to the hub, and start a server with the GPU profile you
+   just set up.
+2. Open a terminal, and try running `nvidia-smi`. This should provide
+   you output indicating that a GPU is present.
+3. Open a notebook, and run the following python code to see if
+   tensorflow can access the GPUs:
+
+   ```python
+   import tensorflow as tf
+   tf.config.list_physical_devices('GPU')
+   ```
+
+   This should output something like:
+
+   ```
+   [PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
+   ```
+4. Remember to explicitly shut down your server after testing,
+   as GPU instances can get expensive!
+
+If either of those tests fail, something is wrong and off you go debugging :)
```

Comment:
In the future, we should develop a GPU debugging guide

Reply:
I made a note in https://github.com/2i2c-org/infrastructure/issues/1167, @sgibson91